head/src/main/kotlin/io/qalipsis/core/head/report/ReportServiceImpl.kt

```diff
@@ -257,7 +268,8 @@ internal class ReportServiceImpl(
             page = reportEntityPage.pageNumber,
             totalPages = reportEntityPage.totalPages,
             totalElements = reportEntityPage.totalSize,
-            elements = reportEntityPage.content.map { reportConverter.convertToModel(it) }
+            elements = reportEntityPage.content.map { id -> reportRepository.findById(id)!! }
```

Comment:
Use `findAllById(reportEntityPage.content)` and reorder them according to the page content to drastically improve the performance.